packages/commonwealth/client/scripts/state/ui/sidebar/sidebar.ts

```diff
@@ -18,12 +23,37 @@ interface SidebarStore {
   }) => void;
 }
 
+const setUserSidebarVisibility = (state: 'open' | 'closed' | null) => {
+  if (state) {
+    localStorage.setItem('user-sidebar-visibility', state);
+  } else {
+    localStorage.removeItem('user-sidebar-visibility');
+  }
+};
+
+const userSidebarPreference = localStorage.getItem('user-sidebar-visibility');
```

Comment:
This is a workaround and we should not use localStorage anymore when using zustand. 
To properly solve this, zustand is providing middleware, in this case, persist middleware, that handles localStorage by default. Note that `devtools` is a middleware, so to make it work, `persist` have to wrap the whole store, similarly to devtools, and as a second argument you need to pass `{name: 'sidebar-store'}` and thats it.

https://docs.pmnd.rs/zustand/recipes/recipes#persist-middleware

Reply:
Is fixing this specific comment blocking on merge @masvelio ?

Reply:
I see that you updated it, thanks. It is really important that we follow new standards for state management and we are going away from workarounds and old state pattern